inference/core/workflows/core_steps/common/query_language/operations/classification_results/base.py

```diff
@@ -27,6 +27,19 @@ def extract_top_class_confidence(prediction: dict) -> Union[float, List[float]]:
     ]
 
 
+def extract_top_class_confidence_single(prediction: dict) -> Union[float, List[float]]:
+    if "confidence" in prediction:
+        return prediction["confidence"]
+    predicted_classes = prediction.get("predicted_classes", [])
+    predicted_confidences = [
+        prediction["predictions"][class_name]["confidence"]
+        for class_name in predicted_classes
+    ]
+    if not predicted_confidences:
+        return 0.0
```

Comment:
I'm returning 0.0 instead of None because subsequent UQL operations (such as top_class_confidence_single >= 0.5) fail if None is returned